app/main.py

```diff
@@ -1,3 +1,73 @@
+from __future__ import annotations
+
+import operator
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: int | float | Distance) -> Distance | TypeError:
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        if isinstance(other, (int, float)):
+            return Distance(self.km + other)
+
+        raise TypeError("Unsupported operand for addition")
+
+    def __iadd__(self, other: int | float | Distance) -> Distance | TypeError:
+        if isinstance(other, Distance):
+            self.km += other.km
+        elif isinstance(other, (int, float)):
+            self.km += other
+        else:
+            raise TypeError(
+                f"Unsupported operand type(s) for +=: /n"
+                f" 'Distance' and '{type(other)}'"
+            )
+        return self
+
+    def __mul__(self, other: int | float) -> Distance | TypeError:
+        if isinstance(other, (int, float)):
+            return Distance(self.km * other)
+        elif isinstance(other, Distance):
+            raise TypeError("Unsupported operation.")
+
+    def __truediv__(self, other: int | float) -> Distance | TypeError:
+        if isinstance(other, (int, float)):
+            return Distance(round(self.km / other, 2))
+        elif isinstance(other, Distance):
```

Comment:
Since we support division only for ints and floats it means that all other types may result in TypeError thus you don't need this condition here.
Check your other methods and fix accordingly where necessary